### ISSUE TITLE
[java] Badly formated UTF-8 report from Chinese text

### ISSUE THREAD

**Issue by cyw3** [pmd/pmd#698]
when i run the pmd5.8.0 with the xml result output report and use this report, i occur a error below:
![image](https://user-images.githubusercontent.com/11549103/32217360-b56020a6-be62-11e7-9c81-0f4a210bb67b.png)

**Comment by cyw3** [comment]
i run pmd in windows7.

**Comment by jsotuyod** [comment]
@cyw3 interesting! This seems like an encoding issue.

I'd like to ask you a little more info to be able to reproduce / fix this.

1. What encoding is the XML using? (you can check it at the top where it says `<?xml version="1.0" encoding="..."`)
2. What encoding are your source files using? Given you work at Tencent, I assume these may be Chinese characters?

If you can, please upload the report XML. Also if you can, upload the reported source file, or at least copy the reported snippet onto a new file (I want a file, and not just the contents pasted onto Github to check on encodings).

Thanks in advanced

**Comment by cyw3** [comment]
1、UTF-8
2、my source files use utf-8. Yes, it's Chinese characters.

**Comment by cyw3** [comment]
[PMDErrorOutput.zip](https://github.com/pmd/pmd/files/1433525/PMDErrorOutput.zip)

**Comment by cyw3** [comment]
The String literal &quot;参数不合法&quot; appears 5 times in this file; the first occurrence is on line 58

**Comment by jsotuyod** [comment]
@cyw3 interesting. The 5 glyphs, under UTF-8 are 15 bytes (3 per glyph), however, on your XML report, there are 20 bytes, 4 per glyph.

All my tests so far, with the provided Chinese text, and UTF-8 encoding work properly. This leads me to believing the issue raises from how the input file is decoded rather than how the report is written.

1. How are you running PMD? Is the **source file encoding** properly set?
   - If using CLI, this is done with `-encoding` / `-e` with the default being utf-8
   - If using Gradle, this will default to the system encoding, can be set by overriding Java's `-Dfile.encoding=...`
   - If using Maven, this is done through the `sourceEncoding` pmd property, or by setting the `project.build.sourceEncoding` property. Defaults to the system encoding.
   - If using Ant, this is done with the `encoding` property with the default being the system encoding.

2. I know I already asked, but are we certain this particular file is properly encoded as UTF-8?

**Comment by cyw3** [comment]
I used i with CLI. And I have used the '-encoding utf-8', but I still occured this problem.

**Comment by jsotuyod** [comment]
@cyw3 I can't reproduce it. I tried PMD 5.8.0 on a UTF-8 encoded Java file with the contents:

```java
public class Test {
    public void test() {
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
        System.out.println("参数不合法");
    }
}
```

And I properly get the report output:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<pmd version="5.8.0" timestamp="2017-11-03T01:17:46.112">
<file name="redacted/src/main/java/Test.java">
<violation beginline="3" endline="3" begincolumn="28" endcolumn="34" rule="AvoidDuplicateLiterals" ruleset="String and StringBuffer" class="Test" method="test" externalInfoUrl="https://pmd.github.io/pmd-5.8.0/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals" priority="3">
The String literal &quot;参数不合法&quot; appears 9 times in this file; the first occurrence is on line 3
</violation>
</file>
</pmd>
```

I can think of no other explanation for your issue other than your source file not being properly encoded in UTF-8, and without access to the source file I can think of nothing else to do.

**Comment by jsotuyod** [comment]
Closing this due to lack of feedback and impossibility of reproducing it.
